src/http_server.cc

```diff
@@ -2910,7 +2910,7 @@ HTTPAPIServer::InferRequestClass::FinalizeResponse(
 
   const char* request_id = nullptr;
   RETURN_IF_ERR(TRITONSERVER_InferenceResponseId(response, &request_id));
-  if ((request_id != nullptr) && (request_id[0] != '\0')) {
+  if ((request_id == nullptr) || (request_id[0] == '\0')) {
```

Comment:
I think this check should've remained `!=` in this spot, we don't need to add an empty or nullptr request_id to the response, right?

I would double check each of these request_id related if statements to see that they make sense in each spot in each of your PRs.

Reply:
Thanks so much for your careful reviews. Everything should be good now.